File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max1 = max2 = nums[0]
        for i in range(1,len(nums)):
            max1 = max(nums[i], max1+nums[i])
            max2 = max(max1, max2)
        return max2

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max1, max2;
        max1 = nums[0];
        max2 =  nums[0];
        
        for(int i = 1; i < nums.length; i++){
            max1 = Math.max(nums[i], nums[i] + max1);
            max2 = Math.max(max1, max2);
        }
        return max2;
        
    }
}